src/V8JsiRuntime.cpp

```diff
@@ -1307,14 +1375,60 @@ jsi::HostFunctionType &V8Runtime::getHostFunction(const jsi::Function &obj) {
   std::abort();
 }
 
+namespace {
+std::string getFunctionName(v8::Isolate* isolate, v8::Local<v8::Function> func) {
+  std::string functionNameStr;
+  v8::Local<v8::String> functionNameV8Str =
+      v8::Local<v8::String>::Cast(func->GetName());
+  int functionNameLength = functionNameV8Str->Utf8Length(isolate);
+  if (functionNameLength > 0) {
+    functionNameStr.resize(functionNameLength);
+    functionNameV8Str->WriteUtf8(isolate, &functionNameStr[0]);
+  }
+
+  if (functionNameV8Str.IsEmpty()) {
+    functionNameV8Str = v8::Local<v8::String>::Cast(func->GetInferredName());
+    functionNameLength = functionNameV8Str->Utf8Length(isolate);
+    if (functionNameLength > 0) {
+      functionNameStr.resize(functionNameLength);
+      functionNameV8Str->WriteUtf8(isolate, &functionNameStr[0]);
+    }
+  }
+
+  if (functionNameV8Str.IsEmpty()) {
+    functionNameStr = "<anonymous>";
+  }
+
+  return functionNameStr;
+}
+}
+
 jsi::Value V8Runtime::call(
     const jsi::Function &jsiFunc,
     const jsi::Value &jsThis,
     const jsi::Value *args,
     size_t count) {
+ 
   _ISOLATE_CONTEXT_ENTER
   v8::Local<v8::Function> func =
       v8::Local<v8::Function>::Cast(objectRef(jsiFunc));
+
+  // TODO: This may be a bit costly when there are large number of JS function
+  // calls from native. Evaluate !

```

Comment:
>Evaluate [](start = 24, length = 8)

Could we comment out this and the traces until we can evaluate the cost?

Reply:
Tudor, I think we should keep it and measure it through our tooling. This is not supposed to be expensive as both calls to get the function name only does lookups. My main concern is many conditional statements. 
Having these traces will really help in analyzing issues in future as these give us the function call context.